docs/language-pack-providers/language-pack-contract.md

```diff
@@ -0,0 +1,94 @@
+- `run` Using the `func run` command to start the image, a controlling process loads the function project into memory and listens on port 8080 for incoming HTTP requests. The process is determined by the Language Pack. For example, a Node.js Language Pack may use `npm start` as the controlling process, while a Go Language Pack may invoke a binary compiled during the `build` phase.
+- `invoke` When an incoming HTTP request is received by the controlling process, the CloudEvent if sent, is unmarshalled, and the function is invoked with the payload received.
+- `response` After a function has been invoked by the invocation framework, the return value is sent to the caller. If the function returns a CloudEvent, the invocation framework should respond to the caller with the CloudEvent unchanged. If the function returns any other data, it is sent to the caller. Function invocation frameworks may each provide their own APIs and specifications to augment a function developer's experience. For example, the function developer may be able to return a structure containing a numeric HTTP response code, HTTP headers, and response data. These APIs and specifications are typically unique to the runtime environment and language, and as such are left to Language Pack implementors to provide and document. API capabilities for built-in `default` Language Pack runtimes are documented in the function templates themselves. 
+
+## Execution Scope
+When `func create` is used to generate a Function project, the Language Pack provides all of the information necessary for the project to be built as an OCI image. Including for example, specifying use of buildpacks or s2i, what builder images should be used, which environment variables are recognized, and more. In some cases, however, it is possible to simulate the containerized runtime environment locally on a developer's laptop. For example, the built-in `default` Node.js templates can be run locally using standard Node.js development tooling, such as `npm install` and `npm run`. Running functions in this way can be quite convenient, but it is important to remember that a Knative Function project is meant to run within a tightly controlled execution space, where the environment is well defined. Not all Language Packs can provide this functionality, and there is no guarantee that the function invocation will be identical in a local environment.
+
+- Install user function dependencies
+- Load a user function
+- Opening an HTTP socket and listening on a port
+- Receive incoming HTTP requests and convert any received CloudEvent into its native form for the runtime
+- 
+
+## Usage
+
+Using external Language Packs is made possible through the `func repository` command, which allows function developers to add and remove Language Packs from their local development environment. For example
+
+```
+❯ func repository add https://github.com/knative-sandbox/func-tastic functastic # Add the func-tastic repo to the local environment 
```

Comment:
IMHO the `func-tastic` repo should be `functastic`.  Has the boat already sailed on that one?

Reply:
> IMHO the `func-tastic` repo should be `functastic`. Has the boat already sailed on that one?

It has - and I think I agree. But at this point, the machinations required to change are too much to deal with IMO.